tests/common/include/tests/common/geometry_barcode.inl

```diff
@@ -0,0 +1,39 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+// Detray include(s)
+#include "detray/geometry/barcode.hpp"
+
+// System include(s)
+#include <cmath>
+
+using namespace detray;
+
+/// Test retrieval of surface from collection using brute force searching
+TEST(geometry, barcode) {
+
+    auto bcd = geometry::barcode{};
+
+    // Check a empty barcode
+    EXPECT_EQ(bcd.volume(), std::pow(2UL, 8UL) - 1UL);
+    EXPECT_EQ(bcd.id(), static_cast<surface_id>(std::pow(2UL, 4UL) - 1UL));
+    EXPECT_EQ(bcd.index(), std::pow(2UL, 44UL) - 1UL);
+    EXPECT_EQ(bcd.extra(), std::pow(2UL, 8UL) - 1UL);
```

Comment:
Be careful! As far as I know, `std::pow` is designed for use with floats, so your result might be imprecise. Recommend replacing `std::pow(2UL, x)` with `1UL << x`.